File: frontend/de/unika/ipd/grgen/ir/expr/map/MapSizeExpr.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 6.0
 * Copyright (C) 2003-2021 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

/**
 * @author Moritz Kroll, Edgar Jakumeit
 */

package de.unika.ipd.grgen.ir.expr.map;

import de.unika.ipd.grgen.ir.expr.Expression;
import de.unika.ipd.grgen.ir.type.basic.IntType;

public class MapSizeExpr extends MapFunctionMethodInvocationBaseExpr
{
	public MapSizeExpr(Expression targetExpr)
	{
		super("map size expression", IntType.getType(), targetExpr);
	}
}

File: frontend/de/unika/ipd/grgen/ir/expr/map/MapSizeExpr.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

/// <summary>
/// @author Moritz Kroll, Edgar Jakumeit
/// </summary>

namespace de.unika.ipd.grgen.ir.expr.map
{
using Expression = de.unika.ipd.grgen.ir.expr.Expression;
using IntType = de.unika.ipd.grgen.ir.type.basic.IntType;

public class MapSizeExpr : MapFunctionMethodInvocationBaseExpr
{
	public MapSizeExpr(Expression targetExpr)
		: base("map size expression", IntType.Type, targetExpr)
	{
	}
}

}
